### ISSUE TITLE
Crash on select with condition and order by clause 

### ISSUE THREAD

**Issue by jans2002** [GreptimeTeam/greptimedb#1750]
### What type of bug is this?

Crash

### What subsystems are affected?

Standalone mode

### What happened?

When I query with the SQL with some where clause and order by clause, it should output selected rows, but it crashed.

### What operating system did you use?

Ubuntu 20.04

### Relevant log output and stack trace

```bash
2023-06-09T01:15:29.414643Z ERROR common_telemetry::panic_hook: panicked at 'index out of bounds: the len is 0 but the index is 1', src/storage/src/read.rs:78:10 backtrace=   0: common_telemetry::panic_hook::set_panic_hook::{{closure}}
             at /home/jason/code/rust/greptimedb/src/common/telemetry/src/panic_hook.rs:31:25
   1: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/alloc/src/boxed.rs:1999:9
      std::panicking::rust_panic_with_hook
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/panicking.rs:695:13
   2: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/panicking.rs:582:13
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/sys_common/backtrace.rs:150:18
   4: rust_begin_unwind
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/panicking.rs:578:5
   5: core::panicking::panic_fmt
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/panicking.rs:67:14
   6: core::panicking::panic_bounds_check
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/panicking.rs:162:5
   7: <usize as core::slice::index::SliceIndex<[T]>>::index
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/slice/index.rs:258:10
      core::slice::index::<impl core::ops::index::Index<I> for [T]>::index
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/slice/index.rs:18:9
      <alloc::vec::Vec<T,A> as core::ops::index::Index<I>>::index
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/alloc/src/vec/mod.rs:2691:9
      storage::read::Batch::column
             at /home/jason/code/rust/greptimedb/src/storage/src/read.rs:78:10
      storage::schema::projected::ProjectedSchema::batch_to_chunk::{{closure}}
             at /home/jason/code/rust/greptimedb/src/storage/src/schema/projected.rs:179:38
      core::iter::adapters::map::map_fold::{{closure}}
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/iter/adapters/map.rs:84:28
      core::iter::traits::iterator::Iterator::fold
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/iter/traits/iterator.rs:2481:21
      <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/iter/adapters/map.rs:124:9
      <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/iter/adapters/map.rs:124:9
      <core::iter::adapters::cloned::Cloned<I> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/iter/adapters/cloned.rs:60:9
      core::iter::traits::iterator::Iterator::for_each
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/iter/traits/iterator.rs:856:9
      alloc::vec::Vec<T,A>::extend_trusted
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/alloc/src/vec/mod.rs:2844:17
      <alloc::vec::Vec<T,A> as alloc::vec::spec_extend::SpecExtend<T,I>>::spec_extend
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/alloc/src/vec/spec_extend.rs:26:9
      <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/alloc/src/vec/spec_from_iter_nested.rs:62:9
      <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/alloc/src/vec/spec_from_iter.rs:33:9
   8: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/alloc/src/vec/mod.rs:2712:9
      core::iter::traits::iterator::Iterator::collect
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/iter/traits/iterator.rs:1895:9
      storage::schema::projected::ProjectedSchema::batch_to_chunk
             at /home/jason/code/rust/greptimedb/src/storage/src/schema/projected.rs:181:18
      <storage::chunk::ChunkReaderImpl as store_api::storage::chunk::ChunkReader>::project_chunk
             at /home/jason/code/rust/greptimedb/src/storage/src/chunk.rs:64:9
   9: <mito::table::MitoTable<R> as table::table::Table>::scan_to_stream::{{closure}}::{{closure}}
             at /home/jason/code/rust/greptimedb/src/mito/src/table.rs:294:33
      <async_stream::async_stream::AsyncStream<T,U> as futures_core::stream::Stream>::poll_next
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/async-stream-0.3.5/src/async_stream.rs:56:13
  10: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/futures-core-0.3.28/src/stream.rs:120:9
      <common_recordbatch::RecordBatchStreamAdaptor as futures_core::stream::Stream>::poll_next
             at /home/jason/code/rust/greptimedb/src/common/recordbatch/src/lib.rs:225:9
  11: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/futures-core-0.3.28/src/stream.rs:120:9
      <common_recordbatch::adapter::DfRecordBatchStreamAdapter as futures_core::stream::Stream>::poll_next
             at /home/jason/code/rust/greptimedb/src/common/recordbatch/src/adapter.rs:98:15
  12: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/futures-core-0.3.28/src/stream.rs:120:9
      futures_util::stream::stream::StreamExt::poll_next_unpin
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/futures-util-0.3.28/src/stream/stream/mod.rs:1632:9
      <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/futures-util-0.3.28/src/stream/stream/next.rs:32:21
      datafusion::physical_plan::repartition::RepartitionExec::pull_from_input::{{closure}}
             at /home/jason/.cargo/git/checkouts/arrow-datafusion-8f64698f94ffb3e6/63e52dd/datafusion/core/src/physical_plan/repartition/mod.rs:517:40
  13: tokio::runtime::task::core::Core<T,S>::poll::{{closure}}
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/core.rs:223:17
      tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/loom/std/unsafe_cell.rs:14:9
  14: tokio::runtime::task::core::Core<T,S>::poll
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/core.rs:212:13
  15: tokio::runtime::task::harness::poll_future::{{closure}}
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/harness.rs:476:19
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/panicking.rs:485:40
      std::panicking::try
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/panicking.rs:449:19
      std::panic::catch_unwind
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/panic.rs:140:14
      tokio::runtime::task::harness::poll_future
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/harness.rs:464:18
      tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/harness.rs:198:27
      tokio::runtime::task::harness::Harness<T,S>::poll
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/harness.rs:152:15
  16: tokio::runtime::task::raw::RawTask::poll
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/raw.rs:200:18
      tokio::runtime::task::LocalNotified<S>::run
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/mod.rs:394:9
      tokio::runtime::scheduler::multi_thread::worker::Context::run_task::{{closure}}
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/scheduler/multi_thread/worker.rs:464:18
      tokio::runtime::coop::with_budget
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/coop.rs:107:5
      tokio::runtime::coop::budget
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/coop.rs:73:5
      tokio::runtime::scheduler::multi_thread::worker::Context::run_task
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/scheduler/multi_thread/worker.rs:463:9
  17: tokio::runtime::scheduler::multi_thread::worker::Context::run
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/scheduler/multi_thread/worker.rs:426:24
  18: tokio::runtime::scheduler::multi_thread::worker::run::{{closure}}
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/scheduler/multi_thread/worker.rs:406:17
      tokio::macros::scoped_tls::ScopedKey<T>::set
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/macros/scoped_tls.rs:61:9
  19: tokio::runtime::scheduler::multi_thread::worker::run
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/scheduler/multi_thread/worker.rs:403:5
  20: tokio::runtime::scheduler::multi_thread::worker::Launch::launch::{{closure}}
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/scheduler/multi_thread/worker.rs:365:45
      <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/blocking/task.rs:42:21
      tokio::runtime::task::core::Core<T,S>::poll::{{closure}}
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/core.rs:223:17
      tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/loom/std/unsafe_cell.rs:14:9
  21: tokio::runtime::task::core::Core<T,S>::poll
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/core.rs:212:13
  22: tokio::runtime::task::harness::poll_future::{{closure}}
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/harness.rs:476:19
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/panicking.rs:485:40
      std::panicking::try
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/panicking.rs:449:19
      std::panic::catch_unwind
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/panic.rs:140:14
      tokio::runtime::task::harness::poll_future
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/harness.rs:464:18
      tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/harness.rs:198:27
      tokio::runtime::task::harness::Harness<T,S>::poll
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/harness.rs:152:15
  23: tokio::runtime::task::raw::RawTask::poll
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/raw.rs:200:18
      tokio::runtime::task::UnownedTask<S>::run
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/task/mod.rs:431:9
      tokio::runtime::blocking::pool::Task::run
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/blocking/pool.rs:159:9
      tokio::runtime::blocking::pool::Inner::run
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/blocking/pool.rs:513:17
  24: tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}
             at /home/jason/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/tokio-1.28.2/src/runtime/blocking/pool.rs:471:13
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/sys_common/backtrace.rs:134:18
  25: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/thread/mod.rs:529:17
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/panicking.rs:485:40
      std::panicking::try
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/panicking.rs:449:19
      std::panic::catch_unwind
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/panic.rs:140:14
      std::thread::Builder::spawn_unchecked_::{{closure}}
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/thread/mod.rs:528:30
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/core/src/ops/function.rs:250:5
  26: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/alloc/src/boxed.rs:1985:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/alloc/src/boxed.rs:1985:9
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/a368898de758e1b8def6c9060044a5b40eb79e84/library/std/src/sys/unix/thread.rs:108:17
  27: start_thread
             at /build/glibc-SzIz7B/glibc-2.31/nptl/pthread_create.c:477:8
  28: clone
             at /build/glibc-SzIz7B/glibc-2.31/misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:95
 panic.file="src/storage/src/read.rs" panic.line=78 panic.column=10
thread 'mysql-io-handlers' panicked at 'index out of bounds: the len is 0 but the index is 1', src/storage/src/read.rs:78:10
```

### How can we reproduce the bug?

The table create SQL (from MySQL client)
```
create table yn_data_env (
    dev_id int,
    dev_type varchar(30),
    tag varchar(30),
    val double,
    ts timestamp,
    time index(ts)
)
```
The insert some data and query with the following SQL
```
select * from yn_data_env
         where ts<cast('2023-06-08 02:10:00' as timestamp)
         order by ts desc
```

**Comment by v0y4g3r** [comment]
This is a known degradation and will be fixed in #1748

**Comment by killme2008** [comment]
Already fixed in #1748 , thanks for your report.
